### PR TITLE
Update log collector unit file to remove memorylimit

### PR DESCRIPTION
**Pull request by maddieford**
## Description
This previous PR (https://github.com/Azure/WALinuxAgent/pull/2637) removed memory limit from the log collector slice unit file, but cgroupconfigurator currently only updates the unit file if it does not already exist. This memory limit change is included in 2.9.0.4. For VMs which upgrade to 2.9.0.4 from earlier versions, the unit file is created with a memorylimit and never updated. 

This PR updates cgroupconfigurator to always update the log collector slice unit file to remove memorylimit. 

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/logcollector.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def _set_resource_usage_cgroups(cpu_cgroup_path, memory_cgroup_path):
     @staticmethod
     def _initialize_telemetry():
         protocol = get_protocol_util().get_protocol(init_goal_state=False)
-        protocol.client.reset_goal_state(goalstate_properties=GoalStateProperties.RoleConfig | GoalStateProperties.HostingEnv)
+        protocol.client.reset_goal_state(goal_state_properties=GoalStateProperties.RoleConfig | GoalStateProperties.HostingEnv)
```
Comment (maddieford) [inline azurelinuxagent/common/logcollector.py:122]: I found this issue from a previous PR because collect-logs was failing to run

File: azurelinuxagent/common/cgroupconfigurator.py [hunk 2]
```diff
@@ -368,10 +368,9 @@ def __setup_azure_slice():
             if not os.path.exists(vmextensions_slice):
                 files_to_create.append((vmextensions_slice, _VMEXTENSIONS_SLICE_CONTENTS))
 
-            if not os.path.exists(logcollector_slice):
-                slice_contents = _LOGCOLLECTOR_SLICE_CONTENTS_FMT.format(cpu_quota=_LOGCOLLECTOR_CPU_QUOTA)
-
-                files_to_create.append((logcollector_slice, slice_contents))
+            ## Log collector slice should be updated to remove MemoryLimit quota from previous GA versions
```
Comment (narrieta) [inline azurelinuxagent/common/cgroupconfigurator.py:371]: I would rephrase this comment a little. In this specific instance we want to remove the limit set by 2.8, but in general what we want is to update the settings with the values used by the current version (which is what this code is doing)

File: tests/common/test_cgroupconfigurator.py [hunk 3]
```diff
@@ -188,6 +188,27 @@ def test_initialize_should_create_unit_files_when_the_agent_service_file_is_not_
...
 
+    def test_initialize_should_update_logcollector_memorylimit(self):
+        with self._get_cgroup_configurator(initialize=False) as configurator:
+            log_collector_unit_file = configurator.mocks.get_mapped_path(UnitFilePaths.logcollector)
+            original_memory_limit = "MemoryLimit=30M"
+
+            # The mock creates the slice unit file with memory limit
+            configurator.mocks.add_data_file(os.path.join(data_dir, 'init', "azure-walinuxagent-logcollector.slice"),
+                                             UnitFilePaths.logcollector)
+            self.assertTrue(os.path.exists(log_collector_unit_file),
```
Comment (narrieta) [inline tests/common/test_cgroupconfigurator.py:199]: setup errors should be reported with exceptions, rather than asserts, to distinguish them from actual test verifications (such as line 208 below)

File: azurelinuxagent/common/cgroupconfigurator.py (hunk 2 above)
Comment (maddieford) [reply azurelinuxagent/common/cgroupconfigurator.py]: Updated the comment

File: tests/common/test_cgroupconfigurator.py (hunk 3 above)
Comment (maddieford) [reply tests/common/test_cgroupconfigurator.py]: Updated this check & the next check that memory limit exists to raise exceptions instead

**Review by narrieta — APPROVED** [review]

**Review by nagworld9 — APPROVED** [review]
